### PR TITLE
GODRIVER-1483 add error labels to WriteException and BulkWriteException

### PR DESCRIPTION
**Pull request by iwysiu**
this will be tested with the RetryableWriteError label work

### CODE REVIEW

File: x/mongo/driver/errors.go [hunk 1]
```diff
@@ -91,6 +92,11 @@ func (wce WriteCommandError) Error() string {
 
 // Retryable returns true if the error is retryable
 func (wce WriteCommandError) Retryable() bool {
+	for _, label := range wce.Labels {
```
Comment (divjotarora) [inline x/mongo/driver/errors.go]: Can you explain this change? On a network error, the wire message sending code always returns an `Error` with the `NetworkError` label. If we got a `WriteCommandError`, that already means the network communication was successful, so we should never hit this case.

Comment (iwysiu) [reply x/mongo/driver/errors.go]: I was pretty sure I didn't need that check, but I wasn't sure so I added it in case. Removed.

**Comment by divjotarora** [comment]
@jyemin the testing would happen in the retryable writes error labels work. The goal was to split this off, merge it, rebase that work, and then have those tests take care of it. @iwysiu Can you confirm?

**Comment by iwysiu** [comment]
@jyemin @divjotarora yup, those would be covered by the retryable writes tests

**Comment by jyemin** [comment]
Can't we rebase the error label work on _this_ branch rather than committing this branch to master and rebasing on master?

**Comment by iwysiu** [comment]
I added a test to make sure that it gets propagated all the way down, which won't be checked by the retryable writes tests.

File: mongo/integration/crud_prose_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	"go.mongodb.org/mongo-driver/internal/testutil/assert"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func TestCrudProse(t *testing.T) {
+	clientOpts := options.Client().SetRetryWrites(false).SetWriteConcern(mtest.MajorityWc).
+		SetReadConcern(mtest.MajorityRc)
+	mtOpts := mtest.NewOptions().ClientOptions(clientOpts).MinServerVersion("3.6").Topologies(mtest.ReplicaSet, mtest.Sharded).
```
Comment (divjotarora) [inline mongo/integration/crud_prose_test.go]: Failpoints don't exist until 4.2 on sharded clusters afaik. I think they're 4.0+ on replica sets.

File: mongo/integration/crud_prose_test.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		CreateClient(false)
+	mt := mtest.New(t, mtOpts)
+	defer mt.Close()
+
+	mt.Run("BulkWriteException with RetryableWriteError label", func(mt *mtest.T) {
```
Comment (divjotarora) [inline mongo/integration/crud_prose_test.go]: Do we want tests for other CRUD methods here too?

File: mongo/integration/crud_prose_test.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		label := "RetryableWriteError"
+		mt.SetFailPoint(mtest.FailPoint{
+			ConfigureFailPoint: "failCommand",
+			Mode: mtest.FailPointMode{
```
Comment (divjotarora) [inline mongo/integration/crud_prose_test.go]: Might need to make this fail twice as the retryable writes code probably already retries on 11600.

Comment (iwysiu) [reply mongo/integration/crud_prose_test.go]: changed the error

File: mongo/integration/crud_prose_test.go (hunk 3 above)
Comment (iwysiu) [reply mongo/integration/crud_prose_test.go]: I added tests that go through the other possible paths

File: mongo/integration/crud_prose_test.go (hunk 2 above)
Comment (iwysiu) [reply mongo/integration/crud_prose_test.go]: done

**Comment by iwysiu** [comment]
will be merged with #287

**Pull request closed without merging**